### PR TITLE
Examples:  Add pie and donut charts

### PR DESCRIPTION
**Pull request by louismaximepiton**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

NA

### Description

Added an example showing a pie and a donut charts.

Questions still to be answered: 
- Do we make several pages for these two kinds of charts or only one is enough ? If not, remove the `ods-donut-charts.png`.
- Not quite sure of the values I gave inside the examples. Need to be discussed or is it fine as a first draft ?
- Do we need to update `getThemeOptions` accordingly ? For example we don't have `axis` on this kind of chart.

### Motivation & Context

These type of charts are considered as basic and are missing inside the doc.

### Types of change

- New feature (non-breaking change which adds functionality)

### Test checklist

Please check that the following tests projects are still working:

- [x] `docs/examples`
- [ ] `test/angular-ngx-echarts`
- [ ] `test/angular-tour-of-heroes`
- [ ] `test/html`
- [ ] `test/react`
- [ ] `test/vue`
- [ ] `test/examples/bar-line-chart`
- [ ] `test/examples/single-line-chart`
- [ ] `test/examples/timeseries-chart`

### CODE REVIEW

**Comment by jacques-lebourgeois** [comment]
Hi, I push 3 commits, 2 suggestion and one enhancement.

Available to discuss about them :)

**Review by louismaximepiton — COMMENTED** [review]
I feel right with your changes, the display is great and as an alpha version it's fine to add some support to legend I think

File: docs/examples/index.js
Comment (jacques-lebourgeois) [reply docs/examples/index.js:379]: I asked myself the question. 

If the aim is to limit specific dev, then it's better to use the default behaviour of Apache Echarts, provided that the design generated with Apache Echarts legends and adapted by ods-charts is considered satisfactory.

- To use ECharts legends: the look&feel is satisfactory
- To use ods-charts legends: behaviour at the limits of Apache Echarts can lead to the internal legends and the graphs being superimposed.

File: docs/examples/index.js
Comment (jacques-lebourgeois) [reply docs/examples/index.js:250]: There are three types of legend to manage.

NB: it should be possible to choose a legend type independently of the graph type, even if there is a default legend type for each graph type.

**Comment by jacques-lebourgeois** [comment]
> #158 has been merged, please apply the same modifications here

Commited

File: docs/examples/index.js
Comment (louismaximepiton) [reply docs/examples/index.js]: Tackled by #244

File: docs/examples/index.js
Comment (louismaximepiton) [reply docs/examples/index.js]: Tackled by #242

**Comment by louismaximepiton** [comment]
Removed the work you did @jacques-lebourgeois since it'll be embedded in #242 and following PRs. I think it's ready for review.

Note: I've added the Orange legend look for now, but it'll have to be removed if the #242 is merged before or if the #242 is merged after, think about reverting https://github.com/Orange-OpenSource/ods-charts/pull/88/commits/780f2cb376e0ed3c0e427cf7150c6ce0d1ad59cd

**Comment by jacques-lebourgeois** [comment]
I just push a commit for suggestion if you agree

**Review by jacques-lebourgeois — APPROVED** [review]

**Comment by nbossard** [comment]
We are waiting for this feature for project Orange Money Pay Metrics...

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Good job with this addition @louismaximepiton 
Few tiny things to change before accepting.
Please check https://github.com/Orange-OpenSource/ods-charts/pull/300 to see if there are other changes to do here while rebasing the branch ;)

**Review by julien-deramond — APPROVED** [review]
LGTM! Let's wait for the light design review, and then we can merge/release 🚀

**Comment by Franco-Riccitelli** [comment]
Here are some design review comments for the Pie chart and Donut chart.
- The use of the "Default" colours looks good to me. I can see these are based on the ODS Categorical colours for data displays.
- The spacing between each of the segments in the charts is also very good.
- The padding on the tooltips is quite large. For the Donut this can sometimes result in the tooltip overlapping the value in the centre of the Donut (when the chart is reduced in size). I suggest trying to reduce the padding in the tooltip by about half and also reducing the vertical space between the text labels in the tooltip by about a half.
- When the charts are reduced in size, I also think the font size can be reduced for the legend and the text in the tooltip. I suggest trying 14px font size.
- The legend should align with the title and subtitle at the top.  
- The wrapping of the legend works well.

**Comment by julien-deramond** [comment]
Thanks for the design review @Franco-Riccitelli 🙏 

> The padding on the tooltips is quite large.
> I suggest trying to reduce the padding in the tooltip by about half and also reducing the vertical space between the text labels in the tooltip by about a half.

The padding on the tooltips follows the standard from ODS. We did not initially notice the different specification hidden at https://design.orange.com/0c1af118d/p/714762-legends/t/page-714762-48288420-5. Should we adhere to this specification instead for all charts? If so, this adjustment can be addressed in a separate PR, as it impacts the global rendering for all charts.

> When the charts are reduced in size, I also think the font size can be reduced for the legend and the text in the tooltip. I suggest trying 14px font size.

We can implement this change in separate PRs since it affects the global rendering for all charts:
* One PR for adjusting the label font size to 14px at small breakpoints
* Another PR (linked to the previous feedback) for tooltip rendering adjustments

However, we need to determine the breakpoint at which the font size changes to 14px. Should this adjustment occur from tablet rendering to mobile (medium)?

> The legend should align with the title and subtitle at the top. At the moment, the legend is positioned slightly to the right.

According to the [ODS Design guidelines](https://design.orange.com/0c1af118d/p/714762-legends/t/page-714762-48288439-8), the legend should be vertical and positioned on the right-hand side of the donut/pie chart. This change is not included in the current PR because it depends on the merge of another PR ([PR #244](https://github.com/Orange-OpenSource/ods-charts/pull/244)). You can preview the updates in [this deployment](https://deploy-preview-244--ods-charts.netlify.app/use_cases/legends-holders).
Do you confirm this must be positioned on the right-hand side of the donut/pie chart?

**Comment by Franco-Riccitelli** [comment]
In response to the questions to my review of the Donut and Pie charts:

The padding for the tooltip should be reduced for all chart usage. I understand this will be a separate PR as this impacts the global rendering, which totally makes sense.

Font size change: I think the 14px font change should occur from the tablet to mobile breakpoints.

Legend positioning: The legend positioning at the bottom of the Pie and Donut chart works well and should remain at the bottom. The only slight adjustment is the alignment of the Legend to the Title and Subtitle (see attached example). Regarding the guidelines, I will update the Data Display Guidelines to show that the legend can be positioned at the bottom of Donut and Pie charts.
Screenshot 2024-07-22 at 10 30 38 AM

**Comment by julien-deramond** [comment]
Thanks a lot for your feedback, @Franco-Riccitelli. 
We've addressed the specific points related to the donut/pie charts in this PR. The more general feedback will be managed separately. We plan to create dedicated issues for those and implement the modifications through separate PRs.

**Comment by nbossard** [comment]
👏 good news

**Pull request merged**
